tests/Datastore/DatastoreSessionHandlerTest.php

```diff
@@ -0,0 +1,514 @@
+<?php
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Tests\Datastore;
+
+use Google\Cloud\Datastore\DatastoreClient;
+use Google\Cloud\Datastore\DatastoreSessionHandler;
+use Google\Cloud\Datastore\Entity;
+use Google\Cloud\Datastore\Key;
+use Google\Cloud\Datastore\Query\Query;
+use Google\Cloud\Datastore\Transaction;
+use InvalidArgumentException;
+
+/**
+ * @group datastore
+ */
+class DatastoreSessionHandlerTest extends \PHPUnit_Framework_TestCase
+{
+    const KIND = 'PHPSESSID';
+    const NAMESPACE_ID = 'sessions';
+
+    private $datastore;
+    private $transaction;
+
+    public function setUp()
+    {
+        $this->datastore = $this->getMockBuilder(DatastoreClient::class)
```

Comment:
The rest of the library uses [Prophesy](https://phpunit.de/manual/current/en/test-doubles.html#test-doubles.prophecy) (bundled with PHPUnit) for test doubles. Can we switch from the PHPUnit mock builder to Prophesy?